paramiko/rsakey.py

```diff
@@ -26,26 +26,31 @@
 from cryptography.hazmat.primitives.asymmetric import rsa, padding
 
 from paramiko.message import Message
-from paramiko.pkey import PKey
+from paramiko.pkey import PKey, register_pkey_type
 from paramiko.py3compat import PY2
 from paramiko.ssh_exception import SSHException
 
 
+@register_pkey_type
 class RSAKey(PKey):
     """
     Representation of an RSA key which can be used to sign and verify SSH2
     data.
     """
 
+    LEGACY_TYPE = 'RSA'
+    OPENSSH_TYPE_PREFIX = 'ssh-rsa'
+
     def __init__(self, msg=None, data=None, filename=None, password=None,
-                 key=None, file_obj=None):
+                 key=None, file_obj=None, raw=None):
```

Comment:
All `*Key` constructors now take a `raw=` argument that's used internally to pass raw data from `load_private_key`.

Should I call this `_raw` to emphasize the fact that it's an internal API?

Reply:
Yeah, could rename to `_raw`.

But here's another idea: take a `file_data` instead, which is the full string. I think users would find it useful. I did say I didn't want to have the parsing code written multiple times, or read the file multiple times ... but I am not against running the parsing on the full string multiple times (once to get type, once to actually decrypt and load). But I guess users could just use the new `pkey.load_private_key()` anyway. So, just an idea.

Reply:
(in this idea I guess the new kwarg would be `key_str` instead of `file_data`, for consistency with `load_private_key()`)

Reply:
Personally I prefer not parsing the data twice. I think users should stop constructing the `*Key` object types directly and instead rely on `load_private_key*()` as well, so improving the old-style API seems like a non-goal.

Reply:
OK, let's call it `_raw` to show that it is internal API. After all, it's not really "raw" like bytes, it's half-processed ...

Reply:
Agreed, I was just making that change as I saw your reply about it :)